KenanLab1.java

```diff
@@ -1,15 +1,101 @@
 import java.awt.event.*;
+import java.util.*;
+import javax.swing.*;
 
 public class KenanLab1 implements KeyListener
 {
+private JFrame frame;
+private JTextField field;
+private JLabel label;
+private JLabel label2;
+
+private int[][] maze;
+private int xLoc, yLoc;
+private ArrayList<String> history;
+private String options;
+private static boolean finished = false;
+
+   public KenanLab1()
+   {
+   frame = new JFrame();
+   frame.setSize(200,200);
+   frame.setLocation(500,300);
+   frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+   buildPanel();
+   
+   history = new ArrayList<String>();
+   maze = new int[16][8];
+   for(int col = 0; col < maze.length; col++)
+      {
+      for(int row = 0; row < maze[col].length; row++)
+         {
+         if(col==0 || col == maze.length || row == 0 || row == maze[col].length)
+            maze[col][row] = 1;
+         }
+      }
+   Random random = new Random();
+   xLoc = random.nextInt(12)+ 2;
+   yLoc = random.nextInt(4) + 2;
+   options = "";
+   frame.setVisible(true);
+   showOptions();
+   }
+   
+   public void buildPanel()
+   {
+   field = new JTextField();
+   field.addKeyListener(this);
+   frame.add(field);
+   label = new JLabel();
+   frame.add(label);
+   label2 = new JLabel();
+   frame.add(label2);
+   }
+   
    public static void main (String[] args)
    {
+   new KenanLab1();

```

Comment:
When naming classes, KenanLab1 is not going to help anyone out. When we get to building basic classes, this will be obvious, but for now, they need to be nouns representing the single use purpose of the class